pkg/compose/cp.go

```diff
@@ -105,6 +87,35 @@ func (s *composeService) Copy(ctx context.Context, projectName string, options a
 	return g.Wait()
 }
 
+func (s *composeService) listContainersTargetedForCopy(ctx context.Context, projectName string, index int, direction copyDirection, serviceName string) (Containers, error) {
+	var containers Containers
+	var err error
+	if direction == fromService || (direction == toService && index > 0) {
```

Comment:
nested `if`s make this a bit hard to read, maybe just split this into simpler top-level `switch` statement?

```go
switch {
	case direction == fromService && index == 0: 
                containers = ...
	case index != 0:
                containers = ...
        default:
                containers = ...
	}
```